fop/fom.c

```diff
@@ -532,7 +533,8 @@ static void thr_addb2_leave(struct m0_loc_thread *thr,
 M0_INTERNAL void m0_fom_wakeup(struct m0_fom *fom)
 {
 	fom->fo_cb.fc_ast.sa_cb = readyit;
-	m0_sm_ast_post(&fom->fo_loc->fl_group, &fom->fo_cb.fc_ast);
+	if (fom->fo_cb.fc_ast.sa_next == NULL)
```

Comment:
Well, it's true that `sa_next` can be NULL here already when the cb() is not executed yet and the ast will be posted again. But in this particular case it's not a problem, because `readyit()` callback will make sure to not call `m0_fom_ready()` when the fop is not in the waiting state anymore.

Reply:
Well, it's true that `sa_next` can be NULL here already when the cb() is not executed yet and the ast will be posted again. But in this particular case it's not a problem, because `readyit()` callback will make sure to not call `m0_fom_ready()` when the fop is not in the waiting state anymore.